src/lib/server/orderTab.ts

```diff
@@ -93,46 +97,94 @@ export async function concludeOrderTab({ slug }: { slug: string }) {
-		_id: line._id.toString(),
-		productId: line.productId,
-		quantity: line.quantity,
-		internalNote: line.internalNote,
-		chosenVariations: line.chosenVariations
-	}));
+
+	// Build cart items
+	const cartItems = itemQuantities
+		? Array.from(itemQuantities.entries()).map(([itemId, qty]) => {
+				const tabItem = orderTab.items.find((i) => i._id.toString() === itemId);
+				if (!tabItem) {
+					const availableItems = orderTab.items.map((i) => i._id.toString());
+					console.error('Item not found in orderTab:', {
+						requestedItemId: itemId,
+						availableItems
+					});
+					throw error(400, 'Item not found in order tab');
+				}
+				return {
+					_id: itemId,
+					productId: tabItem.productId,
+					quantity: qty,
+					internalNote: tabItem.internalNote,
+					chosenVariations: tabItem.chosenVariations
+				};
+		  })
+		: orderTab.items.map((line) => ({
+				_id: line._id.toString(),
+				productId: line.productId,
+				quantity: line.quantity,
+				internalNote: line.internalNote,
+				chosenVariations: line.chosenVariations
+		  }));
+
+	// Determine final splitMode
+	const finalSplitMode = itemQuantities ? 'items' : splitMode;
+
 	const createResult = await collections.carts.insertOne({
 		_id: new ObjectId(),
 		items: cartItems,
 		orderTabSlug: slug,
+		orderTabId: orderTab._id,
+		...(finalSplitMode && { splitMode: finalSplitMode }),
 		updatedAt: new Date(),
 		createdAt: new Date(),
 		user
 	});
-	const cart = await collections.carts.findOne({ _id: createResult.insertedId });
-	if (!createResult.acknowledged || !cart) {
+
+	if (!createResult.acknowledged) {
 		throw new Error('Failed to create cart');
 	}
+
+	// Link items to cartId
 	await collections.orderTabs.updateOne(
 		{ _id: orderTab._id },
 		{ $set: { 'items.$[elem].cartId': createResult.insertedId } },
-		{ arrayFilters: [{ 'elem._id': { $in: cart.items.map((item) => new ObjectId(item._id)) } }] }
+		{ arrayFilters: [{ 'elem._id': { $in: cartItems.map((item) => new ObjectId(item._id)) } }] }
 	);
 }
 
 export async function addToOrderTab(params: {
 	tabSlug: string;
 	productId: string;
-}): Promise<{ success: true } | { success: false; error: string; maxQuantity: number }> {
+}): Promise<{ success: true } | { success: false; error: string; maxQuantity?: number }> {
 	const orderTab = await getOrCreateOrderTab({ slug: params.tabSlug });
 
-	const existingLineIndex = orderTab.items.findIndex(
-		(item) => !item.cartId && !item.orderId && item.productId === params.productId
+	const sharesOrder = await collections.orders.findOne({
+		orderTabId: orderTab._id,
+		splitMode: 'shares',
+		payments: { $elemMatch: { status: 'paid' } }
+	});
+	if (sharesOrder) {
+		return { success: false, error: 'sharesPaymentStarted' };
+	}
+
+	const poolStartedPayments = orderTab.items.some((i) => i.originalQuantity !== undefined);
+	const existingItem = orderTab.items.find(
+		(i) => !i.cartId && i.productId === params.productId && (!i.orderId || i.quantity === 0)
```

Comment:
Hard to read.

The adjacent conditions are not trivial. Break this down into a stream, filter the adjacent conditions and then find what you're looking for.

For example:

```
... = orderTab.items
  // Exclude items already in a cart since [...]
  .filter((i) => !i.cartId)
  // Exclude items already in an order unless [...]
  .filter((i) => !i.orderId || i.quantity == 0)
  .find((i) => i.productId === params.productId)
```